### PR TITLE
CppCheck FaultsManager FluidCooler FluidProperties FuelCellElectricGenerator

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck issues
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/FaultsManager.cc [hunk 1]
```diff
@@ -363,7 +346,8 @@ namespace FaultsManager {
             if (lAlphaFieldBlanks(2)) {
                 state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).AvaiSchedPtr = -1; // returns schedule value of 1
             } else {
-                state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).AvaiSchedPtr = GetScheduleIndex(state, cAlphaArgs(2));
+                state.dataFaultsMgr->FaultsEvapCoolerFouling(jFault_EvapCoolerFouling).AvaiSchedPtr =
```
Comment (amirroth) [inline src/EnergyPlus/FaultsManager.cc]: `auto &` shortcuts would help here

File: src/EnergyPlus/FaultsManager.cc [hunk 2]
```diff
@@ -511,7 +496,7 @@ namespace FaultsManager {
                 // Check whether the chiller name and chiller type match each other
                 ChillerNum = 0;
                 int thisChil = 0;
-                for (auto &ch : state.dataPlantChillers->ElectricChiller) {
+                for (auto const &ch : state.dataPlantChillers->ElectricChiller) {
```
Comment (amirroth) [inline src/EnergyPlus/FaultsManager.cc:494]: `find_if`?

File: src/EnergyPlus/FaultsManager.cc [hunk 3]
```diff
@@ -613,7 +598,7 @@ namespace FaultsManager {
                 // Check whether the chiller name and chiller type match each other
                 ChillerNum = 0;
                 int thisChil = 0;
-                for (auto &ch : state.dataPlantChillers->ConstCOPChiller) {
+                for (auto const &ch : state.dataPlantChillers->ConstCOPChiller) {
```
Comment (amirroth) [inline src/EnergyPlus/FaultsManager.cc:595]: `find_if`?

File: src/EnergyPlus/FaultsManager.cc [hunk 4]
```diff
@@ -2130,17 +2111,14 @@ namespace FaultsManager {
...
         if (ScheduleManager::GetCurrentScheduleValue(state, this->AvaiSchedPtr) > 0.0) {
 
             // Check fault severity schedules (Ptr initialized to -1, so would return a FaultFrac of 1 if not set)
-            FaultFrac = ScheduleManager::GetCurrentScheduleValue(state, this->SeveritySchedPtr);
+            return ScheduleManager::GetCurrentScheduleValue(state, this->SeveritySchedPtr);
         }
 
-        return FaultFrac;
+        return 0, 0;
     }
```
Comment (amirroth) [inline src/EnergyPlus/FaultsManager.cc:2103]: ?

File: src/EnergyPlus/FluidCoolers.cc [hunk 5]
```diff
@@ -104,11 +103,12 @@ FluidCoolerspecs *FluidCoolerspecs::factory(EnergyPlusData &state, DataPlant::Pl
...
     // Now look for this particular fluid cooler in the list
-    for (auto &fc : state.dataFluidCoolers->SimpleFluidCooler) {
-        if (fc.FluidCoolerType == objectType && fc.Name == objectName) {
-            return &fc;
-        }
-    }
+    auto thisObj = std::find_if(
+        state.dataFluidCoolers->SimpleFluidCooler.begin(),
+        state.dataFluidCoolers->SimpleFluidCooler.end(),
+        [&objectType, &objectName](const FluidCoolerspecs &myObj) { return myObj.FluidCoolerType == objectType && myObj.Name == objectName; });
```
Comment (amirroth) [inline src/EnergyPlus/FluidCoolers.cc]: Bind `objectType` by value since it is a scalar.

File: src/EnergyPlus/FluidCoolers.cc [hunk 6]
```diff
@@ -1950,17 +1939,10 @@ void FluidCoolerspecs::update(EnergyPlusData &state)
...
 
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine is for passing results to the outlet water node.
 
-    // SUBROUTINE PARAMETER DEFINITIONS:
-
-    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    Real64 LoopMinTemp;
-
     auto &waterOutletNode = this->WaterOutletNodeNum;
```
Comment (amirroth) [inline src/EnergyPlus/FluidCoolers.cc]: Should be an `int` not an `auto &` (no good way to grep for this pattern).

File: src/EnergyPlus/FluidCoolers.cc [hunk 7]
```diff
@@ -1990,8 +1972,8 @@ void FluidCoolerspecs::update(EnergyPlusData &state)
     }
 
     // Check if OutletWaterTemp is below the minimum condenser loop temp and warn user
-    LoopMinTemp = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).MinTemp;
-    if (this->OutletWaterTemp < LoopMinTemp && this->WaterMassFlowRate > 0.0) {
+    Real64 LoopMinTemp = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).MinTemp;
+    if (this->OutletWaterTemp<LoopMinTemp &&this->WaterMassFlowRate> 0.0) {
```
Comment (amirroth) [inline src/EnergyPlus/FluidCoolers.cc]: `clang-format` gone wild.

File: src/EnergyPlus/FaultsManager.cc (hunk 4 above)
Comment (rraustad) [reply src/EnergyPlus/FaultsManager.cc:2103]: fat fingers, 0.0

File: src/EnergyPlus/FluidProperties.cc [hunk 8]
```diff
@@ -4890,131 +4810,103 @@ namespace FluidProperties {
...
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int RefrigNum;
-        int IndexNum;
-        bool Failure;
-
-        for (RefrigNum = 1; RefrigNum <= state.dataFluidProps->NumOfRefrigerants; ++RefrigNum) {
-            for (IndexNum = 1; IndexNum <= state.dataFluidProps->RefrigData(RefrigNum).NumPsPoints; ++IndexNum) {
+        for (int RefrigNum = 1; RefrigNum <= state.dataFluidProps->NumOfRefrigerants; ++RefrigNum) {
+            for (int IndexNum = 1; IndexNum <= state.dataFluidProps->RefrigData(RefrigNum).NumPsPoints; ++IndexNum) {
```
Comment (amirroth) [inline src/EnergyPlus/FluidProperties.cc]: `auto &refrig` shortcut would simplify

File: src/EnergyPlus/FluidProperties.cc [hunk 9]
```diff
@@ -5068,37 +4958,18 @@ namespace FluidProperties {
...
-        int GlycolIndex;    // index used in routine / function calls, value is returned on first use (when index=0)
 
         state.dataFluidProps->GetInput = false; // input has already been gotten
 
-        for (GlycolNum = 1; GlycolNum <= state.dataFluidProps->NumOfGlycols; ++GlycolNum) {
-            GlycolIndex = 0; // used in routine calls -- value is returned when first 0
+        for (int GlycolNum = 1; GlycolNum <= state.dataFluidProps->NumOfGlycols; ++GlycolNum) {
+            int GlycolIndex = 0; // used in routine calls -- value is returned when first 0
             // Lay out the basic values:
             if (!state.dataFluidProps->GlycolData(GlycolNum).GlycolName.empty()) {
```
Comment (amirroth) [inline src/EnergyPlus/FluidProperties.cc]: `auto &glycol`

File: src/EnergyPlus/FluidProperties.cc [hunk 10]
```diff
@@ -8751,38 +8169,12 @@ namespace FluidProperties {
...
-
-        // DERIVED TYPE DEFINITIONS:
-        // na
-
-        // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int GlycolNo;
-
         // check and see if it might be a refrigerant
-        GlycolNo = FindGlycol(state, RefrigerantName);
+        int GlycolNo = FindGlycol(state, RefrigerantName);
```
Comment (Myoldmopar) [inline src/EnergyPlus/FluidProperties.cc:8177]: Every bit of cleanup in fluid properties is greatly appreciated.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Found an unused variable from a previous merge and cleaned that up.  This is otherwise all clean, and ready to go.  Thanks @rraustad

**Pull request merged**